org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/editor/folding/FoldingRange.java

```diff
@@ -0,0 +1,113 @@
+/*******************************************************************************
+ * Copyright (c) 2021 TypeFox GmbH (http://www.typefox.io) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.xtext.ide.editor.folding;
+
+import org.eclipse.xtext.util.ITextRegion;
+
+import com.google.common.base.Objects;
+
+/**
+ * @author Mark Sujew - Initial contribution and API
```

Comment:
Please add a comment indicating the reason why this type is present, e.g. abstraction over FoldedPosition and LSP FoldingRange